crates/circuit/src/symbol_expr.rs

```diff
@@ -0,0 +1,3214 @@
+            Value::Int(e) => Value::Real((*e as f64).asin()),
+            Value::Complex(e) => {
+                let t = Value::Complex(e.asin());
+                match t.opt_complex() {
+                    Some(v) => v,
+                    None => t,
+                }
+            }
+        }
+    }
+    pub fn cos(&self) -> Value {
+        match self {
+            Value::Real(e) => Value::Real(e.cos()),
+            Value::Int(e) => Value::Real((*e as f64).cos()),
+            Value::Complex(e) => {
+                let t = Value::Complex(e.cos());
+                match t.opt_complex() {
+                    Some(v) => v,
+                    None => t,
+                }
+            }
+        }
+    }
+    pub fn acos(&self) -> Value {
+        match self {
+            Value::Real(e) => Value::Real(e.acos()),
+            Value::Int(e) => Value::Real((*e as f64).acos()),
+            Value::Complex(e) => {
+                let t = Value::Complex(e.acos());
+                match t.opt_complex() {
+                    Some(v) => v,
+                    None => t,
+                }
+            }
+        }
+    }
+    pub fn tan(&self) -> Value {
+        match self {
+            Value::Real(e) => Value::Real(e.tan()),
+            Value::Int(e) => Value::Real((*e as f64).tan()),
+            Value::Complex(e) => {
+                let t = Value::Complex(e.tan());
+                match t.opt_complex() {
+                    Some(v) => v,
+                    None => t,
+                }
+            }
+        }
+    }
+    pub fn atan(&self) -> Value {
+        match self {
+            Value::Real(e) => Value::Real(e.atan()),
+            Value::Int(e) => Value::Real((*e as f64).atan()),
+            Value::Complex(e) => {
+                let t = Value::Complex(e.atan());
+                match t.opt_complex() {
+                    Some(v) => v,
+                    None => t,
+                }
+            }
+        }
+    }
+    pub fn exp(&self) -> Value {
+        match self {
+            Value::Real(e) => Value::Real(e.exp()),
+            Value::Int(e) => Value::Real((*e as f64).exp()),
+            Value::Complex(e) => {
+                let t = Value::Complex(e.exp());
+                match t.opt_complex() {
+                    Some(v) => v,
+                    None => t,
+                }
+            }
+        }
+    }
+    pub fn log(&self) -> Value {
+        match self {
+            Value::Real(e) => Value::Real(e.ln()),
+            Value::Int(e) => Value::Real((*e as f64).ln()),
+            Value::Complex(e) => {
+                let t = Value::Complex(e.ln());
+                match t.opt_complex() {
+                    Some(v) => v,
+                    None => t,
+                }
+            }
+        }
+    }
+    pub fn sqrt(&self) -> Value {
+        match self {
+            Value::Real(e) => Value::Real(e.sqrt()),
+            Value::Int(e) => {
```

Comment:
what happens if e is a real negative number? then the output should be complex and not real?